src/pages/_app.js

```diff
@@ -1,27 +1,45 @@
 import { ThemeProvider } from 'styled-components';
 import theme from '../styles/theme';
 import GlobalStyle from '../styles/global-style';
-import Navbar from '../components/Navbar';
+import Navbar from '../components/Menu/Navbar';
 import { useRouter } from 'next/router';
 import { SessionProvider } from 'next-auth/react';
 import ProtectLayouts from '../components/ProtectLayouts';
+import { useState } from 'react';
+import Sidebar from '../components/Menu/Sidebar';
+import { Box } from '@mui/material';
 
 function MyApp({ Component, pageProps, session }) {
     const router = useRouter();
 
+    const [open, setOpen] = useState(false);
+    const [selectedLink, setSelectedLink] = useState(null);
+
+    const handleDrawerOpen = () => {
+        setOpen(true);
+    };
+
     return (
         <SessionProvider session={session}>
             <ThemeProvider theme={theme}>
-                {router.pathname !== '/login' ? (
+                {router.pathname !== '/loginPage' ? (
                     <ProtectLayouts>
                         <GlobalStyle />
-                        <Navbar></Navbar>
-                        <Component {...pageProps} />{' '}
+                        <Box sx={{ display: 'flex' }}>
+                            <Navbar {...{ open, setOpen, handleDrawerOpen }} />
+                            <Sidebar {...{ open, setOpen, selectedLink }}>
```

Comment:
useState의 set 값을 바로 전달하는것은 별로 추천하지 않는 방식입니다.
handleOpen 값을 만들어 사용하면 handleDrawerOpen에서도 사용하고 아래 setOpen대신 사용하실 수 있을 것같아요
무엇보다도 boolean 을 단순히 다루는 함수라면 useBoolean이나 useToggle 같은 custom hook을 만들어 사용하시길 추천드립니다.

>It is not recommended to pass the set value of useState immediately.
If you make and use the handleOpen value, you can use it in the handleDrawerOpen and use it instead of the setOpen below
Above all, if it is a function that simply deals with boolean, I recommend that you make a custom hook such as useBoolean or useToggle.